brian2/core/variables.py

```diff
@@ -844,6 +844,15 @@ def set_item(self, item, value, level=0, namespace=None):
         if variable.read_only:
             raise TypeError('Variable %s is read-only.' % self.name)
 
+        # Check whether the group allows writing to the variable (e.g. for
+        # synaptic variables, writing is only allowed after a connect)
+        try:
+            self.group.check_variable_write(variable)
+        except ReferenceError:
+            # Ignore problems with weakly referenced groups that don't exist
+            # anymore at this time (e.g. when doing neuron.axon.var = ...)
+            pass
```

Comment:
In what sense should it result in an error? Writing `neuron.axon.var = ...` works fine, but the `VariableView` for "var" does only hold a weak reference to the `SpatialNeuronSubgroup` that is generated on-the-fly for `neuron.axon`. We had a long discussions about `VariableView` holding a strong or weak reference to its source at some point (discussed mostly in the context of a monitors, though), and we decided to go with weak references, so I don't see anything obvious to change here.